services/gateway/gateway-service.mjs

```diff
@@ -245,53 +250,84 @@ class GatewayService {
    * - displayName: The name to show for this player.
    * - displayChar: The character asset to display.
    */
-  async onUserLocation(socket, msg) {
+  async onUserLocation(socket, origMsg) {
+    const msg = {};
     msg.playerID = socket.playerID;
     msg.displayName = socket.playerData.displayName;
     msg.displayChar = socket.playerData.displayChar;
+    msg.facing = origMsg.facing;
+    msg.mapCoord = MapCoord.fromObject(origMsg.mapCoord);
     // TODO: Check if facing is valid.
     // TODO: Check if movement is legal.
 
-    let mapSize = this.gameMap.getMapSize();
-    let lastMsg = { x:socket.playerData.x , y:socket.playerData.y}
-    if(lastMsg.x === undefined || lastMsg.y === undefined){
-      lastMsg.x = 0;
-      lastMsg.y = 0;
+    let lastCoord = socket.playerData.mapCoord;
+    let mapSize = this.gameMap.getMapSize(lastCoord.mapName);
+    if(lastCoord.x === undefined || lastCoord.y === undefined || lastCoord.mapName === undefined){
+      // Shouldn't happen, log an error.
+      console.error(`Invalid lastCoord in onUserLocation ${lastCoord}`);
+      return;
```

Comment:
Do we really need this check?

Reply:
This is a guard so that we'll know when we screw up playerData.

Reply:
I think we should use `assert` instead of `console.error` if our intention is to know when we screw up something.
```
assert('x' in lastCoord && 'y' in lastCoord && 'mapName' in lastCoord);
```

Reply:
I am wondering that `socket.playerData.mapCoord` is guaranteed to be a `class MapCoord` in `GatewayService.addSocket`. I mean, a crash of the server is reasonable if we really mess up something. We should not handle this error if this is not allowed to happen.